addons/sale_stock/models/sale_order.py

```diff
@@ -192,6 +193,92 @@ class SaleOrderLine(models.Model):
     product_packaging = fields.Many2one('product.packaging', string='Package', default=False)
     route_id = fields.Many2one('stock.location.route', string='Route', domain=[('sale_selectable', '=', True)], ondelete='restrict')
     move_ids = fields.One2many('stock.move', 'sale_line_id', string='Stock Moves')
+    product_type = fields.Selection(related='product_id.type')
+    qty_at_date = fields.Float(compute='_compute_qty_at_date')
+    scheduled_date = fields.Datetime(compute='_compute_qty_at_date')
+    qty_available = fields.Float(compute='_compute_qty_at_date')
+    qty_on_hand = fields.Float(compute='_compute_qty_at_date')
+    warehouse_view_id = fields.Many2one(related='order_id.warehouse_id.view_location_id')
+    qty_to_deliver = fields.Float(compute='_compute_qty_to_deliver')
```

Comment:
i really think you should use the same name of the ocmpute fields and append _at_date or today

qty_available_at_date
free_qty_today
virtual_available_at_date

else it is really difficult to match the mentally to the original compute